documentation/docs/validator-zone/external-chains/polygon.md

```diff
@@ -0,0 +1,184 @@
+```
+
+##### 2. Install `golang`
+
+```bash
+wget https://gist.githubusercontent.com/ssandeep/a6c7197811c83c71e5fead841bab396c/raw/go-install.sh
+bash go-install.sh
+sudo ln -nfs ~/.go/bin/go /usr/bin/go
+```
+
+##### 2. Install `RabbitMq`
+
+```bash
+docker run -it --rm --name rabbitmq -p 5672:5672 -p 15672:15672 rabbitmq:3.9-management
+```
+
+## Install the Polygon node
+Polygon node consists of 2 layers, `Heimdall` and `Bor`. Heimdall is a fork of tendermint and is running in parallel to an Ethereum network, monitoring contracts. Bor is a fork of go-Ethereum and producing blocks shuffled by Heimdall nodes. You need to install and run both binaries in the correct order as explained in the following steps. 
+
+##### 1. Install Heimdall 
+Please make sure you checkout the [latest release tag](https://github.com/maticnetwork/heimdall/tags). In this tutorial we are using `v0.2.5-beta1`
+```bash
+cd ~/
+git clone https://github.com/maticnetwork/heimdall
+cd heimdall
+git checkout v0.2.5-beta1
+make install
+
+# Verify the correct version
+heimdalld version --long
+```
+
+##### 2. Install Bor
+Please make sure you checkout the [latest release tag](https://github.com/maticnetwork/bor/tags). In this tutorial we are using `v0.2.13-beta2`
+```bash
+cd ~/
+git clone https://github.com/maticnetwork/bor
+cd bor
+git checkout v0.2.13-beta2
+make bor-all
+sudo ln -nfs ~/bor/build/bin/bor /usr/bin/bor
+sudo ln -nfs ~/bor/build/bin/bootnode /usr/bin/bootnode
+
+# Verify the correct version
+bor version 
+```
+## Setup and configure node
+
+##### 1. Setup launch directory
+
+```bash
+cd ~/
+git clone https://github.com/maticnetwork/launch
+
+mkdir -p node
+cp -rf launch/testnet-v4/sentry/sentry/* ~/node
+```
+
+##### 2. Setup network directories
+
+```bash
+# Heimdall
+cd ~/node/heimdall
+bash setup.sh
+
+# Bor
+cd ~/node/bor
+bash setup.sh
+```
+
+##### 3. Setup service files
+
+```bash
+# Download service file
+cd ~/node
+wget https://raw.githubusercontent.com/maticnetwork/launch/master/testnet-v4/service.sh
+
+# Generate Metadata
+sudo mkdir -p /etc/matic
+sudo chmod -R 777 /etc/matic/
+touch /etc/matic/metadata
+
+# Generate service file and copy them into systemd directory
+cd ~/node
+bash service.sh
+sudo cp *.service /etc/systemd/system/
+```
+
+##### 4. Setup config files
+
+Open the `~/.heimdalld/config/config.toml` and edit:
+```bash
+moniker=<enter unique identifier>
+seeds="4cd60c1d76e44b05f7dfd8bab3f447b119e87042@54.147.31.250:26656,b18bbe1f3d8576f4b73d9b18976e71c65e839149@34.226.134.117:26656"
+```
+
+Open the `~/.heimdalld/config/heimdall-config.toml` and edit:
+```bash
+eth_rpc_url = <insert Infura or any full node RPC URL to Goerli>
+```
+
+Open the `~/node/bor/start.sh` and add the following flag to start parameters:
+```bash
+--bootnodes "enode://320553cda00dfc003f499a3ce9598029f364fbb3ed1222fdc20a94d97dcc4d8ba0cd0bfa996579dcc6d17a534741fb0a5da303a90579431259150de66b597251@54.147.31.250:30303"
+```
+
+## Start the Polygon services
```

Comment:
Polygon has snapshot for testnets available here: https://snapshots.matic.today/ which can save hours if not days of sync up

Reply:
see previous reply.